test/apps/test_ucx_tls.py

```diff
@@ -140,6 +169,59 @@ def test_fallback_from_rc(dev, neps) :
 
     os.unsetenv("UCX_TLS")
 
+def test_ucx_tls_positive(tls):
+    # Use TLS list in "allow" mode and verify that the found tl is in the list
+    found_tl = find_am_transport(None, tls=tls)
+    print "Using UCX_TLS=" + tls + ", found TL: " + str(found_tl)
+    if tls == 'all':
+        return
+    tls = tls.split(',')
+    if found_tl in tls:
+        return
+    for tl in tls:
+        if tl in tl_aliases:
+            if found_tl in tl_aliases[tl]:
+                return
+    sys.exit(1)
+
+def test_ucx_tls_negative(tls):
+    # Use TLS list in "negate" mode and verify that the found tl is not in the list
+    found_tl = find_am_transport(None, tls="^"+tls)
+    print "Using UCX_TLS=^" + tls + ", found TL: " + str(found_tl)
+    tls = tls.split(',')
+    if not found_tl or found_tl in tls:
+        sys.exit(1)
+    for tl in tls:
+        if tl in tl_aliases:
+            if found_tl in tl_aliases[tl]:
+                sys.exit(1)
+
+def _powerset(iterable, with_empty_set=True):
+    iterable_list = list(iterable)
+    return itertools.chain.from_iterable(
+        itertools.combinations(iterable_list, r) for r in \
+            range(0 if with_empty_set else 1, len(iterable_list) + 1))
+
+def test_tls_allow_list(ucx_info):
+    status, output = exec_cmd(ucx_info + " -d | grep Transport | awk '{print $3}'")
+    available_tls = set(output.splitlines())
+
+    # Add some basic variants (those that are available on this platform)
+    tls_variants = [tls_variant for tls_variant in ["tcp", "posix", "xpmem"] if \
+                    tls_variant in available_tls]
+
+    # Add some IB variant (both strict and alias), if available
+    for tls_variant in available_tls:
+        if tls_variant.startswith("rc_") or tls_variant.startswith("dc_") or \
+           tls_variant.startswith("ud_"):
+            tls_variants += ["ib", tls_variant]
```

Comment:
when using TL from "ucx_info -d", better pass `\\rc_mlx5` for example, in case alias has the same name as a transport